code/modules/station_goals/bluespace_tap.dm

```diff
@@ -255,6 +255,61 @@
 	if(!powernet)
 		connect_to_network()
 
+/obj/machinery/power/bluespace_tap/update_icon(updates)
+	. = ..()
+	overlays.Cut()
+	underlays.Cut()
+	if(length(active_nether_portals))
+		icon_state = "redspace_tap"
+		overlays += icon(icon, "redspace")
+		set_light(15, 5, "#ff0000")
+		return
+	if(stat & (BROKEN|NOPOWER))
+		set_light(0)
+	else
+		set_light(1, 1, "#353535")
+
+	if(get_available_power() <= 0)
+		icon_state = base_icon_state
+	else
+		icon_state = "[base_icon_state][get_icon_state_number()]"
+		overlays += icon(icon, "screen")
+		if(light)
+			underlays += emissive_appearance(icon, "light_mask")
```

Comment:
This should probably be broken into the new specific procs (`update_icon_state()` and `update_overlays()`) with underlay logic included in the overlays proc.